### PR TITLE
WMS getstyle

### PR DESCRIPTION
**Pull request by pblottiere**
## Description

By working on https://github.com/qgis/QGIS/pull/41376, I noticed that the WMS `GetStyle` code could be cleaned and improved by using the WMS context.

### CODE REVIEW

**Comment by pblottiere** [comment]
Hello @dmarteau,

Could you take a quick look please?

File: src/server/services/wms/qgswmsgetstyles.cpp [hunk 1]
```diff
@@ -37,52 +38,33 @@ namespace QgsWms
...
   }
 
   namespace
   {
-    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project,
-        QStringList &layerList )
+    QDomDocument getStyledLayerDescriptorDocument( QgsServerInterface *serverIface, const QgsProject *project, const QgsServerRequest &request )
     {
+      // init WMS parameters and context
+      const QgsWmsParameters parameters( QUrlQuery( request.url() ) );
```
Comment (dmarteau) [inline src/server/services/wms/qgswmsgetstyles.cpp:63]: @pblottiere 

Why not using directly `const QgsWmsParameters parameters( request.serverParameters() );` instead the implicit conversion from `QUrlQuery` ?

Comment (dmarteau) [reply src/server/services/wms/qgswmsgetstyles.cpp:63]: Hi @pblottiere 

Looks good to me. See my comment above.

Comment (pblottiere) [reply src/server/services/wms/qgswmsgetstyles.cpp]: > Why not using directly const QgsWmsParameters parameters( request.serverParameters() ); instead the implicit conversion from QUrlQuery ?

Done

**Comment by pblottiere** [comment]
> Looks good to me

Thanks for the review :+1:. I was especially wondering about the `getStyle(s)` method signature and the unused `version` parameter. Happy to clean that part then :).

**Pull request merged**